packages/ckeditor5-table/src/commands/mergecellscommand.js

```diff
@@ -69,10 +70,21 @@ export default class MergeCellsCommand extends Command {
 				}
 			}
 
-			if ( emptyRowsIndexes.length ) {
-				const table = findAncestor( 'table', firstTableCell );
+			const table = findAncestor( 'table', firstTableCell );
 
-				emptyRowsIndexes.reverse().forEach( row => tableUtils.removeRows( table, { at: row, batch: writer.batch } ) );
+			if ( emptyRowsIndexes.length ) {
+				emptyRowsIndexes.reverse().forEach( row => {
+					tableUtils.removeRows( table, { at: row, batch: writer.batch } );
+				} );
+			} else {
+				// If there were some rows removed then empty columns were already verified.
+				const emptyColumnsIndexes = getEmptyColumnsIndexes( table );
+
+				if ( emptyColumnsIndexes.length ) {
+					emptyColumnsIndexes.reverse().forEach( column => {
+						tableUtils.removeColumns( table, { at: column, batch: writer.batch } );
+					} );
+				}
```

Comment:
AFAICS:

1.  This is commonly used together.
2.  Finding empty rows is cheap and can be done without `emptyRowIndexes` in above algorithm.
3.  Part of fixing logic is used as util and part is here.

I think it would be beneficial to check if this logic can be merged to one: `removeEmptyRowsColomuns` or two (`removeEmptyRows()` and `removeEmptyColumns()` ). This way:

*   it is a bit better to have whole "cleanup" logic in one place - it is used this way in the code
*   no need to think how to use `getEmptyColumnIndexes()`
*   the logic of removing empty rows can be in near removing empty columns.

Depending on the outcome, name, etc I would also think where to put this function(s) - `structure.js` is OK.

Reply:
ps.: when documenting new funcitons link them together and write when to use which (if needed).